Sources/Datadog/Core/Utils/EncodableValue.swift

```diff
@@ -61,7 +61,33 @@ internal struct JSONStringEncodableValue: Encodable {
             // Switch to encode `url.absoluteString` directly - see the comment in `EncodableValue`
             try urlValue.absoluteString.encode(to: encoder)
         } else {
-            let jsonData = try jsonEncoder.encode(encodable)
+            let jsonData: Data
+
+            if #available(iOS 13.0, *) {
+                jsonData = try jsonEncoder.encode(encodable)
+            } else {
+                // Prior to `iOS13.0` the `JSONEncoder` is unable to encode primitive values - it expects them to be
+                // wrapped inside top-level JSON object (array or dictionary). Reference: https://bugs.swift.org/browse/SR-6163
+                //
+                // As a workaround, we serialize the `encodable` as a JSON array and then remove `[` and `]` bytes from serialized data.
+                let temporaryJsonArrayData = try jsonEncoder.encode([encodable])
+
+                let subdataStartIndex = temporaryJsonArrayData.startIndex.advanced(by: 1)
+                let subdataEndIndex = temporaryJsonArrayData.endIndex.advanced(by: -1)
```

Comment:
how does this work? 
i assume `advanced(by: 1)` means advancing 1 byte, how do we know "[/]" is always 1 byte? 
it must be `utf8` encoding for them to be 1 byte, where does `utf8` encoding come from?

Reply:
is this part of JSON spec?

Reply:
Yes, `UTF-8` is part of JSON standard ([RFC-8259](https://tools.ietf.org/html/rfc8259)). Whenever we speak about JSON, we talk about `UTF-8`.